migrations/4_genesis.js

```diff
@@ -6,28 +6,56 @@ const Controller = artifacts.require("Controller")
 const ManagerProxy = artifacts.require("ManagerProxy")
 const LivepeerToken = artifacts.require("LivepeerToken")
 const GenesisManager = artifacts.require("GenesisManager")
-const TokenDistributionMock = artifacts.require("TokenDistributionMock")
+const MerkleMine = artifacts.require("MerkleMine")
+
+const getCurrentBlockTimestamp = async () => {
+    const block = await new Promise((resolve, reject) => {
+        return web3.eth.getBlock("latest", (err, blk) => {
+            if (err) {
+                reject(err)
+            } else {
+                resolve(blk)
+            }
+        })
+    })
+
+    return block.timestamp
+}
 
 module.exports = function(deployer, network, accounts) {
     deployer.then(async () => {
         const lpDeployer = new ContractDeployer(deployer, Controller, ManagerProxy)
 
         const controller = await Controller.deployed()
-
         const tokenAddr = await controller.getContract(contractId("LivepeerToken"))
         const minterAddr = await controller.getContract(contractId("Minter"))
-
         const token = await LivepeerToken.at(tokenAddr)
 
-        const dummyTokenDistribution = await lpDeployer.deploy(TokenDistributionMock, genesis.dummyTokenDistribution.timeToEnd)
-
         let genesisManager
 
+        const grantsStartTimestamp = (await getCurrentBlockTimestamp()) + genesis.timeToGrantsStart
+
         if (!lpDeployer.isProduction(network)) {
-            // If not in production, send the company supply to the deployment account instead of the bank multisig
-            genesisManager = await lpDeployer.deploy(GenesisManager, tokenAddr, dummyTokenDistribution.address, accounts[0], minterAddr)
+            // If not in production, send the crowd supply to the faucet and the company supply to the deployment account
+            deployer.logger.log("Not in production - crowd supply will be sent to faucet and company supply will be sent to deployment account")
+
+            const faucetAddr = await controller.getContract(contractId("LivepeerTokenFaucet"))
+            genesisManager = await lpDeployer.deploy(GenesisManager, tokenAddr, faucetAddr, accounts[0], minterAddr, grantsStartTimestamp)
```

Comment:
When we are not in a production network (i.e. Rinkeby dry run or mainnet), the faucet becomes the token distribution contract and the deployment account becomes the bank